src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -68,18 +69,38 @@ public function getEnv($prefix, $name, \Closure $getEnv)
                 throw new RuntimeException(sprintf('Resolved value of "%s" did not result in an array value.', $next));
             }
             if (!array_key_exists($key, $array)) {
-                throw new RuntimeException(sprintf('Key "%s" not found in "%s" (resolved from "%s")', $key, json_encode($array), $next));
+                throw new EnvNotFoundException(sprintf('Key "%s" not found in "%s" (resolved from "%s")', $key, json_encode($array), $next));
             }
 
             return $array[$key];
         }
 
+        if ('catch' === $prefix) {
+            $i = strrpos($name, ':');
+            if (false === $i) {
+                throw new RuntimeException(sprintf('Invalid configuration: env var "fallback:%s" does not contain a default value.', $name));
+            }
+
+            $next = substr($name, 0, $i);
+            $default = substr($name, $i + 1);
```

Comment:
perhaps check to be non empty, im also worried it causes confusion as the last segment will always be used. If you forget it, another part will be implicitly used.

Reply:
refactored the logic to `default:fallback:origin` in order to catch only `NotFoundException` and let `RuntimeException` bubble